controllers/volumereplication_controller.go

```diff
@@ -196,14 +196,39 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		replicationErr = r.markVolumeAsPrimary(instance, logger, volumeHandle, parameters, secret)
 
 	case replicationv1alpha1.Secondary:
-		replicationErr = r.markVolumeAsSecondary(instance, logger, volumeHandle, parameters, secret)
-		// resync volume if successfully marked Secondary
-		if replicationErr == nil {
-			err := r.updateReplicationStatus(instance, logger, getReplicationState(instance), "volume is marked secondary")
-			if err != nil {
-				return ctrl.Result{}, err
+		// For the first time, Mark the volume as secondary and requeue the
+		// request, For some storage providers it takes some time to determine
+		// whether the volume need correction example:- correcting split brain.
+		if instance.Status.State == "" || instance.Status.State != replicationv1alpha1.SecondaryState {
```

Comment:
is not only second condition sufficient here

Reply:
No, this is to handle the VR creation with secondary replicationState

Reply:
still, 2nd condition should be enough IMO `instance.Status.State != replicationv1alpha1.SecondaryState`

If you have a new CR with the `""` state and only the above condition it will go inside condition as the state is not equal to secodary.

Reply:
Yes looks like the second condition  is enough, fixed it now